src/cr/cube/matrix/cubemeasure.py

```diff
@@ -68,6 +78,164 @@ def _slice_idx_expr(cls, cube, slice_idx):
         return np.s_[slice_idx]
 
 
+# === MEANS ===
+
+
+class _BaseCubeMeans(_BaseCubeMeasure):
+    """Base class for mean cube-measure variants."""
+
+    def __init__(self, dimensions, means):
+        super(_BaseCubeMeans, self).__init__(dimensions)
+        self._means = means
+
+    @classmethod
+    def factory(cls, cube, dimensions, slice_idx):
+        """Return _BaseCubeMeans subclass instance appropriate to `cube`."""
+        dimension_types = cube.dimension_types[-2:]
+        CubeMeansCls = (
+            _MrXMrCubeMeans
+            if dimension_types == (DT.MR, DT.MR)
+            else _MrXCatCubeMeans
+            if dimension_types[0] == DT.MR
+            else _CatXMrCubeMeans
+            if dimension_types[1] == DT.MR
+            else _CatXCatCubeMeans
+        )
+        if cube.means is None:
+            raise ValueError("cube-result does not contain cube-means measure")
```

Comment:
This guard should probably go right up top in this method since none of the rest makes any sense if there is no `cube.means`.